### PR TITLE
Dispatcher -> AsyncDispatcher

### PR DESCRIPTION
**Pull request by tomchristie**
Refactoring for re-introducing the sync API.

### CODE REVIEW

File: httpx/client.py [hunk 1]
```diff
@@ -621,9 +621,9 @@ def redirect_stream(
 
         return response
 
-    def dispatcher_for_url(self, url: URL) -> Dispatcher:
+    def AsyncDispatcher_for_url(self, url: URL) -> AsyncDispatcher:
```
Comment (florimondmanca) [inline httpx/client.py]: This was probably not intended, was it? :)

File: httpx/client.py [hunk 2]
```diff
@@ -889,16 +889,16 @@ def dispatcher_for_url(self, url: URL) -> Dispatcher:
         await self.aclose()
 
 
-def _proxies_to_dispatchers(
+def _proxies_to_AsyncDispatchers(
```
Comment (florimondmanca) [inline httpx/client.py]: Same here - not intended?

File: httpx/client.py [hunk 3]
```diff
@@ -917,15 +917,15 @@ def _proxy_from_url(url: URLTypes) -> Dispatcher:
...
         return {"all": _proxy_from_url(proxies)}
-    elif isinstance(proxies, Dispatcher):
+    elif isinstance(proxies, AsyncDispatcher):
         return {"all": proxies}
     else:
         new_proxies = {}
-        for key, dispatcher_or_url in proxies.items():
-            if isinstance(dispatcher_or_url, (str, URL)):
-                new_proxies[str(key)] = _proxy_from_url(dispatcher_or_url)
+        for key, AsyncDispatcher_or_url in proxies.items():
```
Comment (florimondmanca) [inline httpx/client.py]: Another instance :)

File: httpx/client.py (hunk 1 above)
Comment (tomchristie) [reply httpx/client.py]: oh man.

**Comment by tomchristie** [comment]
Thanks @florimondmanca! 😅

**Comment by tomchristie** [comment]
Right, better now. Thanks for the eyes-on!

**Pull request merged**